### PR TITLE
[Leo] Remove 'New Chat' menu from kebab icon menu in Leo page

### PR DESCRIPTION
**Pull request by jagadeshjai**
Resolves https://github.com/brave/brave-browser/issues/39931

# Outcome

[remove_new_chat_menu.webm](https://github.com/user-attachments/assets/da7d868b-11bc-468a-9a39-fddc7143facb)

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by jagadeshjai** [comment]
@mattmcalister Could you please take a look?

**Comment by mkarolin** [comment]
@jagadeshjai could you, please, rebase. There appears to be a conflict. Thanks.

**Comment by jagadeshjai** [comment]
> @jagadeshjai could you, please, rebase. There appears to be a conflict. Thanks.

 I have resolved the conflicts using the `Resolve Conflict` option on GitHub. Is there still an issue?

**Comment by mkarolin** [comment]
@jagadeshjai, yeah, I was trying to  get the CI running on the PR and it's still telling me there's a conflict:

```
Switched to branch 'contributor-jagadeshjai-remove__new_chat_menu'
+ git fetch origin master
From https://github.com/brave/brave-core
 * branch                    master     -> FETCH_HEAD
+ git rebase FETCH_HEAD
Auto-merging components/ai_chat/core/browser/constants.cc
Auto-merging components/ai_chat/resources/page/components/feature_button_menu/index.tsx
CONFLICT (content): Merge conflict in components/ai_chat/resources/page/components/feature_button_menu/index.tsx
Auto-merging components/resources/ai_chat_ui_strings.grdp
Rebasing (1/1)
error: could not apply 42eb6df2d26... [Leo] Remove 'New Chat' menu from kebab icon menu in Leo page
```
I am not sure if this requires a manual rebase + push on your side or this is an issue with our script. cc: @wknapik

**Comment by jagadeshjai** [comment]
> @jagadeshjai, yeah, I was trying to get the CI running on the PR and it's still telling me there's a conflict:

Yes, I got the conflict in local as well. So for now rebased manually and pushed.
Could you try the CI now? @mkarolin

**Comment by jagadeshjai** [comment]
> All CI checks passed

Thanks @wknapik :smile:

File: components/ai_chat/core/browser/constants.cc [hunk 1]
```diff
@@ -40,7 +40,6 @@ base::span<const webui::LocalizedString> GetLocalizedStrings() {
       {"modelPremiumLabelNonPremium",
        IDS_CHAT_UI_MODEL_PREMIUM_LABEL_NON_PREMIUM},
       {"modelCategory-chat", IDS_CHAT_UI_MODEL_CATEGORY_CHAT},
-      {"menuNewChat", IDS_CHAT_UI_MENU_NEW_CHAT},
```
Comment (petemill) [inline components/ai_chat/core/browser/constants.cc:43]: We can't remove this string - it's still used by `error_conversation_end.tsx`

**Review by petemill — CHANGES_REQUESTED** [review]

File: components/ai_chat/core/browser/constants.cc (hunk 1 above)
Comment (jagadeshjai) [reply components/ai_chat/core/browser/constants.cc:43]: It seems `error_conversation_end.tsx` uses the locale from `src/brave/components/ai_chat/resources/page/stories/locale.ts`, that's why I removed it here.
@petemill

Comment (jagadeshjai) [reply components/ai_chat/core/browser/constants.cc:43]: Okay, Got it. Actually I thought `error_conversation_end.tsx` only uses locale from here https://github.com/brave/brave-core/blob/9303bae9fe07bc2ec7c0cdef139e87304b2d6c96/components/ai_chat/resources/page/stories/locale.ts#L28.

Is it necessary to have these entries in two different files? Why can't we just maintain it in `.grd` file?

@petemill

Comment (jagadeshjai) [reply components/ai_chat/core/browser/constants.cc:43]: Just now I seen that it is under `stories` directory :grin: , so probably used only by storybook :+1:

**Comment by jagadeshjai** [comment]
@petemill I have resolved the comment, PTAL.

**Review by petemill — APPROVED** [review]

**Comment by bsclifton** [comment]
CI looks good! Merging

**Comment by jagadeshjai** [comment]
> CI looks good! Merging

Thanks @bsclifton :smile:

**Pull request merged**